src/main/java/org/fcrepo/spec/testsuite/crud/Container.java

```diff
@@ -35,16 +36,53 @@
 public class Container extends AbstractTest {
 
     /**
-     * 3.1.1-A
+     * 3.1.1-A-1
      */
     @Test(groups = {"MUST"})
     public void createLDPC() {
-        final TestInfo info = setupTest("3.1.1-A",
+        final TestInfo info = setupTest("3.1.1-A-1",
                                         "Implementations must support the creation and management of [LDP] Containers.",
                                         "https://fcrepo.github.io/fcrepo-specification/#ldpc", ps);
         createBasicContainer(uri, info);
     }
 
+    /**
+     * 3.1.1-A-2
+     */
+    @Test(groups = {"MAY"})
+    public void createLDPDirectContainer() {
+        final TestInfo info = setupTest("3.1.1-A-2",
+                                        "Implementations may support the creation and management of [LDP] Direct " +
+                                        "Containers",
+                                        "https://fcrepo.github.io/fcrepo-specification/#ldpc", ps);
+        skipIfDirectContainersNotSupported();
+    }
+
+    /**
+     * 3.1.1-A-3
+     */
+    @Test(groups = {"MAY"})
+    public void createLDPIndirectContainer() {
+        final TestInfo info = setupTest("3.1.1-A-3",
+                                        "Implementations may support the creation and management of [LDP] Indirect " +
+                                        "Containers",
+                                        "https://fcrepo.github.io/fcrepo-specification/#ldpc", ps);
+        skipIfIndirectContainersNotSupported();
+    }
+
+    private void skipIfDirectContainersNotSupported() {
+        if (clientErrorRange().matches(createDirectContainerUnverifed(uri, "").getStatusCode())) {
```

Comment:
For clarity, please add a small dummy body, such as `"some-rdf"`

Reply:
👍